docarray/array/array_stacked.py

```diff
@@ -59,40 +59,43 @@ class DocumentArrayStacked(AnyDocumentArray):
     """
 
     document_type: Type[BaseDocument] = AnyDocument
+    _docs: DocumentArray
 
-    def __init__(self: T, docs: DocumentArray):
-        self._docs = docs
+    def __init__(
+        self: T,
+        docs: DocumentArray,
+    ):
+        self._columns: Dict[str, Union['TorchTensor', T, NdArray]] = {}
 
-        self._columns: Dict[
-            str, Union['TorchTensor', T, NdArray]
-        ] = self._create_columns(docs)
+        self.from_document_array(docs)
 
     def from_document_array(self: T, docs: DocumentArray):
         self._docs = docs
-        self._columns = self._create_columns(docs)
+        self.tensor_type = self._docs.tensor_type
+        self._columns = self._create_columns(docs, tensor_type=self.tensor_type)
 
     @classmethod
     def _create_columns(
-        cls: Type[T], docs: DocumentArray
+        cls: Type[T], docs: DocumentArray, tensor_type: Type['AbstractTensor']
     ) -> Dict[str, Union['TorchTensor', T, NdArray]]:
 
         columns_fields = list()
         for field_name, field in cls.document_type.__fields__.items():
             field_type = field.type_
             if is_union_type(field_type):
-                # cannot stack union fields (might be different types)
-                continue
-
-            is_torch_subclass = (
-                issubclass(field_type, torch.Tensor) if torch_imported else False
-            )
+                if field.type_ == AnyTensor:
```

Comment:
what about other Union types that are not `AnyTensor`? I think it should still work, we just need to check that the provided `tensor_type` is on of the args of the Union

Reply:
You mean if I had a document like this:
```python
class MyDoc(BaseDocument):
    tensor: Union[TorchTensor, NdArray, JaxTensor, SamiTensor]
```

the tensor field should be stacked if I do `DocumentArray(..., tensor_types=TorchTensor)` ?

Reply:
actually I don't agree

Reply:
lets say I have a document like this

```python
class MyDoc(BaseDocument):
    data: Union[TorchTensor, bytes]
```

here I really want to express that my data can either be a tensor or a bytes, therefore I should not be able to stack since the data is not homogeneous. The case of `AnyTensor` is arguably different because it is more of a "mode" and user will either use torch or numpy not both at the same time, but I believe it cannot be extended to every Union otherwise we lose the meaning of what is a Union.

Reply:
but only the templating solution will fix this problem actually

Reply:
ok let's keep it like this for now, and revisit the templating idea at some point in the future